dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -252,12 +254,17 @@ private static ImmutableList<LimitedChannel> createHostChannels(
                 LimitedChannel limitedChannel;
                 if (cf.isConcurrencyLimitingEnabled()) {
                     Channel unlimited = channel;
+                    EndpointChannelState endpointChannelState = channelState.getState(EndpointChannelState.KEY);
                     channel = new ChannelToEndpointChannel(endpoint -> {
                         if (endpoint.tags().contains("dialogue-disable-endpoint-concurrency-limiting")) {
                             return unlimited;
                         }
                         LimitedChannel limited = ConcurrencyLimitedChannel.createForEndpoint(
-                                unlimited, cf.channelName(), uriIndexForInstrumentation, endpoint);
+                                unlimited,
+                                cf.channelName(),
+                                uriIndexForInstrumentation,
+                                endpoint,
+                                endpointChannelState.get(endpoint));
```

Comment:
I think this will fail in an edge case where the endpoint limiter is completely full from a previous refresh, so the queued channel created on the next line cannot sense the boundary condition to begin allowing requests through